central/group/datastore/datastore_impl.go

```diff
@@ -378,3 +392,17 @@ func (ds *dataStoreImpl) getDefaultGroupForProps(ctx context.Context, props *sto
 	// 2. Filter for the default group.
 	return ds.getByProps(ctx, &storage.GroupProperties{AuthProviderId: props.GetAuthProviderId()})
 }
+
+func (ds *dataStoreImpl) validateMutableGroupID(ctx context.Context, id string, force bool) error {
```

Comment:
We should note in a comment that this function assumes the function call is behind a lock, similar to the `...NoLock` functions we already have within this datastore.